auth0/authentication/back_channel_login.py

```diff
@@ -34,4 +34,5 @@ def back_channel_login(
                 "scope": scope,
                 **kwargs,
             },
+            headers={"Content-Type": "application/x-www-form-urlencoded"},
```

Comment:
@kishore7snehil:

The PR's title is _"Adding Support For CIBA with RAR"_, but there’s actually nothing related to RAR. It works simply because the `back_channel_login` method accepts any arguments via `**kwargs`.

What’s the reason for not defining `authorization_details` as an optional parameter in the `back_channel_login` method, similar to how `binding_message`, `scope`, etc, are handled?

As shown in your tests, `authorization_details` is a complex object. Explicitly defining its expected attributes and types would help developers avoid mistakes and also assist in discovering which fields are required.

Reply:
@kishore7snehil: At the very least, consider including it as an expected argument in `back_channel_login` (e.g., authorization_details: Any = None) with proper documentation, as we do in other SDKs. This way, developers are aware that the SDK supports it.

Reply:
> Introducing strict validation or internal structure enforcement would increase the surface area we’d need to maintain and support, which isn't the goal at this layer of the SDK.

Please correct me if I'm wrong, but the `type (string)` attribute is mandatory: https://datatracker.ietf.org/doc/html/rfc9396#name-request-parameter-authoriza